experimental/traffic-portal/src/app/core/users/roles/detail/role-detail.component.html

```diff
@@ -0,0 +1,37 @@
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<mat-card>
+	<tp-loading *ngIf="!role"></tp-loading>
+	<form ngNativeValidate (ngSubmit)="submit($event)" *ngIf="role">
+		<mat-card-content>
+			<mat-form-field>
+				<mat-label>Name</mat-label>
+				<input matInput type="text" name="name" required [(ngModel)]="role.name" [readonly]="role.name === 'admin'"/>
+			</mat-form-field>
+			<mat-form-field>
+				<mat-label>Description</mat-label>
+				<input matInput type="text" name="description" [(ngModel)]="role.description" [readonly]="role.name === 'admin'" />
+			</mat-form-field>
+			<mat-form-field>
+				<mat-label>Permissions</mat-label>
+				<textarea matInput name="permissions" [(ngModel)]="permissions" (input)="updatePermissions()" [readonly]="role.name === 'admin'" cdkTextareaAutosize></textarea>
```

Comment:
instead of doing this on `(input)` - every time the user presses a key - you can just wait until the form is submitted to set the Role's `permissions` based on this text box.

Reply:
Is there something wrong with how I have it currently?